cmd/podman/containers/run.go

```diff
@@ -148,6 +148,33 @@ func run(cmd *cobra.Command, args []string) error {
 		imageName = name
 	}
 
+	// If this is a checkpoint image, invoke container restore.
+	// when checkpointImage == false, err could be "image is not a checkpoint".
+	checkpointImage, err := utils.IsCheckpointImage(registry.GetContext(), []string{imageName})
+	if checkpointImage {
```

Comment:
This check is confusing?  Why do we only return `err` if `checkpointImage` is set?